src/core/utils/apicontext.js

```diff
@@ -0,0 +1,21 @@
+export function isValidContext (context) {
+  // should be both valid JSON and a map
+  let parsed;
+  try {
+    parsed = JSON.parse(context);
+  } catch (e) {
+    return false;
+  }
+
+  if (Array.isArray(parsed)) {
```

Comment:
Is it sufficient to check:
a) JSON.parse works
b) return parsed && typeof parsed === 'Object'?

Reply:
typeof [] is 'object' because javascript. think this can definitely still be simplified though, I think ideally the whole method is within the try catch?

Reply:
Because of short circuiting, would need to check `parsed` alone. But like the idea of the allowlist and shortening the return!